File: lib/domain/entities/order.dart
enum OrderStatus { pending, confirmed, inProduction, completed, cancelled }

class Order {
  final String id;
  final String orderCode;
  final String productId;
  final String productName;
  final String? customerName;
  final String? customerPhone;
  final OrderStatus status;
  final String productionStatus;
  final double totalAmount;
  final double depositAmount;
  final String? note;
  final DateTime createdAt;
  final DateTime? updatedAt;
  final DateTime? estimatedDate;
  final String? assignedStaffName;

  const Order({
    required this.id,
    required this.orderCode,
    required this.productId,
    required this.productName,
    this.customerName,
    this.customerPhone,
    required this.status,
    this.productionStatus = '',
    required this.totalAmount,
    this.depositAmount = 0,
    this.note,
    required this.createdAt,
    this.updatedAt,
    this.estimatedDate,
    this.assignedStaffName,
  });
}

File: backend/src/main/java/com/quyenauto/order/entity/Order.java
package com.quyenauto.order.entity;

import com.quyenauto.common.entity.BaseEntity;
import com.quyenauto.product.entity.Product;
import com.quyenauto.user.entity.User;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "orders")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Order extends BaseEntity {

    @Column(name = "order_code", nullable = false, unique = true, length = 30)
    private String orderCode;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "quotation_id")
    private Quotation quotation;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "customer_id", nullable = false)
    private User customer;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id")
    private Product product;

    @Column(name = "product_name", length = 200)
    private String productName;

    @Builder.Default
    @Column(name = "total_amount", nullable = false, precision = 15, scale = 2)
    private BigDecimal totalAmount = BigDecimal.ZERO;

    @Builder.Default
    @Column(name = "deposit_amount", nullable = false, precision = 15, scale = 2)
    private BigDecimal depositAmount = BigDecimal.ZERO;

    @Builder.Default
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private OrderStatus status = OrderStatus.PENDING;

    @Builder.Default
    @Column(name = "production_status", nullable = false, length = 50)
    private String productionStatus = "NOT_STARTED";

    @Column(columnDefinition = "TEXT")
    private String note;

    @Column(name = "estimated_date")
    private LocalDate estimatedDate;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "assigned_staff_id")
    private User assignedStaff;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL, orphanRemoval = true)
    @OrderBy("createdAt DESC")
    @Builder.Default
    private List<OrderStatusLog> statusLogs = new ArrayList<>();

    public enum OrderStatus {
        PENDING, CONFIRMED, IN_PRODUCTION, COMPLETED, CANCELLED
    }
}
